ruby/lib/gilded_rose.rb

```diff
@@ -1,56 +1,72 @@
 class GildedRose
+  attr_reader :items
 
   def initialize(items)
     @items = items
   end
 
-  def update_quality()
-    @items.each do |item|
-      if item.name != "Aged Brie" and item.name != "Backstage passes to a TAFKAL80ETC concert"
-        if item.quality > 0
-          if item.name != "Sulfuras, Hand of Ragnaros"
-            item.quality = item.quality - 1
-          end
-        end
-      else
-        if item.quality < 50
-          item.quality = item.quality + 1
-          if item.name == "Backstage passes to a TAFKAL80ETC concert"
-            if item.sell_in < 11
-              if item.quality < 50
-                item.quality = item.quality + 1
-              end
-            end
-            if item.sell_in < 6
-              if item.quality < 50
-                item.quality = item.quality + 1
-              end
-            end
-          end
-        end
-      end
-      if item.name != "Sulfuras, Hand of Ragnaros"
-        item.sell_in = item.sell_in - 1
+  def update_quality
+    items.each do |item|
+      update_item(item)
+    end
+  end
+
+  def update_item(item)
+    if item.name == "Sulfuras, Hand of Ragnaros"
+    elsif item.name == "Aged Brie"
+      aged_brie(item)
+    elsif item.name == "Backstage passes to a TAFKAL80ETC concert"
+      backstage_passes(item)
```

Comment:
Refactor to use constants at the top of the class instead of strings.

Reply:
added constants